pkgs/tools/misc/gtkclipblock/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, meson
+, cmake
+, ninja
+, pkg-config
+, gtk2
+, gtk3
+, gtk4
+}:
+let
+  distorm-src = fetchFromGitHub {
+    owner = "gdabah";
+    repo = "distorm";
+    rev = "3.5.2b";
+    hash = "sha256-2ftEV3TMS3HT7f96k+Pwt3Mm31fVEXcHpcbbz05jycU=";
+  };
+  source = fetchFromGitHub {
+    owner = "notpeelz";
+    repo = "gtkclipblock";
+    rev = "v0.2.2";
+    hash = "sha256-ok/D7M0KekN8zf8AzhcOLtedbYVRHHv3m9zEHsJfcPM=";
+    fetchSubmodules = true;
+  };
+in
+stdenv.mkDerivation {
+  pname = "gtkclipblock";
+  version = "0.2.2";
+  src = source;
+  nativeBuildInputs = [ meson cmake ninja pkg-config ];
+  buildInputs = [ gtk2 gtk3 gtk4 ];
+  postPatch = ''
+    substituteInPlace subprojects/funchook-helper/subprojects/funchook/CMakeLists.txt \
+      --replace "GIT_REPOSITORY https://github.com/gdabah/distorm.git" "SOURCE_DIR ${distorm-src}"
```

Comment:
A subproject that gits a subproject that gits a subproject.

How can such a chain work?

Reply:
`gtkclipblock` is a meson project with a git submodule to `funchook-helper` in its `subprojects` folder (`subprojects/funchook-helper`). `funchook-helper` is another meson project (hosted in a different repo) with a git submodule to `funchook` in *its* `subprojects` folder (`subprojects/funchook`). `funchook` is a CMake project that uses FetchContent to get `distorm` (the fourth git repo involved), hence the patch that redirects it to a fetchFromGitHub (because FetchContent can't fetch it when sandboxed).